src/vs/workbench/contrib/extensions/electron-browser/extensionTipsService.ts

```
@@ -1040,4 +1043,77 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
+						Severity.Info,
+						localize('ExtensionsRecommended', "Azure Data Studio has extension recommendations."),
+						[{
+							label: localize('installAll', "Install All"),
+							run: () => {
+								this.telemetryService.publicLog(scenarioType + 'Recommendations:popup', { userReaction: 'install' });
+								const installAllAction = this.instantiationService.createInstance(InstallRecommendedExtensionsByScenarioAction, scenarioType, recommendations);
+								installAllAction.run();
+								installAllAction.dispose();
+
+								if (scenarioType === 'VisualizerExtensions') {
+									c(undefined);
+									const message = 'The  extensions are ready. Select the Visualizer icon to visualize your data.';
+									this.notificationService.info(message);
+								}
+							}
+						}, {
+							label: localize('showRecommendations', "Show Recommendations"),
+							run: () => {
+								this.telemetryService.publicLog(scenarioType + 'Recommendations:popup', { userReaction: 'show' });
+								const showAction = this.instantiationService.createInstance(ShowRecommendedExtensionsByScenarioAction, scenarioType);
+								showAction.run();
+								showAction.dispose();
+								c(undefined);
+							}
+						}, {
+							label: choiceNever,
+							isSecondary: true,
+							run: () => {
+								this.telemetryService.publicLog(scenarioType + 'Recommendations:popup', { userReaction: 'neverShowAgain' });
+								this.storageService.store(storageKey, true, StorageScope.GLOBAL);
+								c(undefined);
+							}
+						}],
+						{
+							sticky: true,
+							onCancel: () => {
+								this.telemetryService.publicLog(scenarioType + 'Recommendations:popup', { userReaction: 'cancelled' });
+								c(undefined);
+							}
+						}
+					);
+				});
+			} else {
+				return Promise.resolve();
+			}
+		});
+	}
+
+	getRecommendedExtensionsByScenario(scenarioType: string): Promise<IExtensionRecommendation[]> {
+		if (!scenarioType) { return Promise.resolve([]); }
```

reject the promise instead of returning empty array

---

When I reject a promise, should a create an error message, such as 
`Promise.reject(new Error("There are no recommended extensions at this time."));`

---

you can say: the scenario type must be provided.